langgraph/src/graph/state.ts

```diff
@@ -374,3 +429,16 @@ export class CompiledStateGraph<
     }
   }
 }
+
+function isStateDefinition(obj: unknown): obj is StateDefinition {
+  return (
+    typeof obj === "object" &&
+    obj !== null &&
+    !Array.isArray(obj) &&
+    Object.keys(obj).length > 0 &&
+    Object.values(obj).every(
+      // eslint-disable-next-line no-instanceof/no-instanceof
+      (v) => typeof v === "function" || v instanceof BaseChannel
```

Comment:
nit: Use presence/absence of `fromCheckpoint` or `lc_graph_name` instead to avoid bundler issues?